java/src/shared/models/Map.java

```diff
@@ -95,21 +173,27 @@ public boolean canSoldier(Soldier_Input params)
 	 */
 	public boolean canBuildRoad(BuildRoad_Input params)
 	{
+		boolean can = false;
 		if(roads.containsKey(params.getRoadLocation())) //already a road here man
 		{
 			return false;
 		}
+		else if(!roadIsOnMap(params.getRoadLocation()))
+		{
+			return false;
+		}
+		
```

Comment:
First if-statement checks if someone has a road built there. Second checks if that is a valid location. For example that would return false if you attempted to build a road between two ocean hexes